python/src/block/SecondaryParticleInformationBlock.python.cpp

```diff
@@ -0,0 +1,96 @@
+// system includes
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+// local includes
+#include "ACEtk/block/SecondaryParticleInformationBlock.hpp"
+#include "views.hpp"
+#include "definitions.hpp"
+
+// namespace aliases
+namespace python = pybind11;
+
+namespace block {
+
+void wrapSecondaryParticleInformationBlock( python::module& module, python::module& ) {
+
+  // type aliases
+  using Block = njoy::ACEtk::block::SecondaryParticleInformationBlock;
+
+  // wrap views created by this block
+
+  // create the block
+  python::class_< Block > block(
+
+    module,
+    "SecondaryParticleInformationBlock",
+    "The continuous energy NTRO block with the number of reactions for each "
+    "secondary particle type\n\n"
+    "The SecondaryParticleInformationBlock class contains the number of reactions\n"
+    "that produce each secondary particle type as given on the PTYPE.\n"
+    "The number of available available particle types is stored in NXS(7)."
+  );
+
+  // wrap the block
+  block
+  .def(
+
+    python::init< std::vector< unsigned int > >(),
+    python::arg( "numbers" ),
+    "Initialise the block\n\n"
+    "Arguments:\n"
+    "    self       the block\n"
+    "    numbers    the number of reaction for each particle type"
+  )
+  .def_property_readonly(
+
+    "NTYPE",
+    &Block::NTYPE,
+    "The number of secondary particle types (excluding the projectile)"
```

Comment:
Since this can only appear in the CE table, this would be excluding the projectile and photons. The other blocks can however appear in both so it depends on the table type: for CE tables it'd be excluding the projectile and photons while for photonuclear it would be excluding the projectile only.